Reason: opt-in wall-clock phase diagnostics for native open-world stage loading
--- a/src/System/MarDirectorLoadResource.cpp
+++ b/src/System/MarDirectorLoadResource.cpp
@@ -1,3 +1,4 @@
+#include <sms_open_world.h>
 #include <System/MarDirector.hpp>
 #include <JSystem/JParticle/JPAEmitterManager.hpp>
 #include <JSystem/JKernel/JKRMemArchive.hpp>
@@ -31,6 +32,7 @@
 
 int TMarDirector::loadResource()
 {
+	sms_open_world_profile("loadResource begin");
 	gpMarioParticleManager = new TMarioParticleManager;
 
 	int particleNum = 1000;
@@ -76,6 +78,7 @@
 	gpEmitterManager4D2
 	    = new JPAEmitterManager(nullptr, 200, 0x20, 0x40, nullptr);
 	loadParticle();
+	sms_open_world_profile("particles ready");
 
 	void* rawArch = SMSLoadArchive("/data/yoshi.arc", nullptr, 0, nullptr);
 	JKRMemArchive* arch = newMemArchive();
@@ -103,7 +106,9 @@
 	if (!paramsArch->mountFixed(paramsBlob, MBF_0))
 		return 1;
 
+	sms_open_world_profile("stage archive begin");
 	unkB8 = gpApplication.mountStageArchive();
+	sms_open_world_profile("stage archive ready");
 	if (!unkB8)
 		return 1;
 
@@ -119,7 +124,9 @@
 #endif
 	unkD4 = new (0x20) char[0x64000];
 	unkD8 = newMemArchive();
+	sms_open_world_profile("thp begin");
 	int errc = thpInit();
+	sms_open_world_profile("loadResource end");
 	if (errc)
 		return errc;
 
--- a/src/System/MarDirectorSetupObjects.cpp
+++ b/src/System/MarDirectorSetupObjects.cpp
@@ -1,2 +1,3 @@
+#include <sms_open_world.h>
 #include <System/MarDirector.hpp>
 #include <JSystem/JKernel/JKRFileFinder.hpp>
@@ -185,4 +186,5 @@
 bool TMarDirector::setupObjects()
 {
+	sms_open_world_profile("setupObjects begin");
 #ifdef TARGET_PC
 	// BetterSunshineEngine (word at retail 0x802B7720): a nop replaces this
@@ -357,4 +359,5 @@
 	    new TSnapTimeObj(0xFFFFFFFF, "Particle Draw SnapTime"));
 
+	sms_open_world_profile("common objects ready");
 	gpConductor->makeGraphGroup(JKRGetResource("/scene/map/scene.ral"));
 
@@ -376,4 +379,5 @@
 
 	{
+		sms_open_world_profile("scene objects begin");
 		void* scene = JKRFileLoader::getGlbResource("/scene/map/scene.bin");
 		u32 size    = unkB8->getResSize(scene);
@@ -410,4 +414,5 @@
 	}
 
+	sms_open_world_profile("scene objects ready");
 	unk80 = new JDrama::TViewObjPtrListT<JDrama::TViewObj>("イベントグループ");
 	root->insert(unk80);
@@ -420,5 +425,7 @@
 	((JKRMemArchive*)JKRFileLoader::getVolume("params"))->unmountFixed();
 	JKRHeap::getCurrentHeap()->freeTail();
+	sms_open_world_profile("loadAfter begin");
 	sceneCommon->loadAfter();
+	sms_open_world_profile("loadAfter end");
 
 	gpConductor->initGraphGroup();
@@ -550,5 +557,7 @@
 	mPerformListCalcAnim->push_back(dialogueCursor, CUE_CALC_ANIM);
 	mPerformListCalcAnim->push_back(targetArrow, CUE_CALC_ANIM);
+	sms_open_world_profile("console begin");
 	setupPerformList_console();
+	sms_open_world_profile("console ready");
 	mPerformListGXPost->push_back(drawInit, CUE_DRAW_STAGE_END);
 	preEntry(unk34);
